lib/spack/env/cc

```diff
@@ -142,6 +151,12 @@ if [[ -z $mode ]]; then
     done
 fi
 
+# Set up rpath variable according to language.
+eval rpath=\$SPACK_${comp}_RPATH_ARG
+if [[ $rpath != *%s* ]]; then
+    die "SPACK_${comp}_RPATH_ARG must contain '%s' for parameter substitution."
```

Comment:
Ah, I see. I think for now it might be better to keep it simple and easy to understand. Otherwise, someone might forget the `%s` and wonder why it isn't working. Hopefully NAG is the only compiler that needs to be coddled like this.